File: Solution/0026.Remove_Duplicates_from_Sorted_Array/0026.Remove_Duplicates_from_Sorted_Array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if(nums.size()==0)
            return 0;
        int i=0,j=1;
        while(j<nums.size())
        {
            if(nums[j]!=nums[i])
            {
                ++i;
                if(j-i>0)        //优化部分：增加原地复制的判断
                    nums[i]=nums[j];
            }
            ++j;
        }
        return i+1;
    }
};

File: Solution/0026.Remove_Duplicates_from_Sorted_Array/0026.Remove_Duplicates_from_Sorted_Array.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        if len(nums)==0:
            return 0
        i,j=0,1
        while j<len(nums):
            if nums[j]!=nums[i]:
                i=i+1
                if i!=j:
                    nums[i]=nums[j]
            j=j+1
        return i+1
                
